packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -109,9 +109,11 @@ class GradleUtils {
       },
     );
     // Add the `gradle-wrapper.properties` file if it doesn't exist.
-    final File propertiesFile = directory.childFile(
-        globals.fs.path.join('gradle', 'wrapper', 'gradle-wrapper.properties'));
+    final Directory propertiesDirectory = directory.childDirectory(
+        globals.fs.path.join('gradle', 'wrapper'));
+    final File propertiesFile = propertiesDirectory.childFile('gradle-wrapper.properties');
     if (!propertiesFile.existsSync()) {
+      propertiesDirectory.createSync(recursive: true);
```

Comment:
I think this was a real bug: I was getting test failures because `directory/gradle/wrapper` didn't exist when this method was called right after the directory was created:
https://github.com/flutter/flutter/blob/b9274c7c607bdb287d3b0483c5a3a8dcf5b35fcb/packages/flutter_tools/lib/src/cache.dart#L1040-L1043

Reply:
nice!